### ISSUE TITLE
Markdown in displayName

### ISSUE THREAD

**Issue by say5** [tektoncd/dashboard#3804]
# Feature request

Support Markdown in `displayName` field.

# Use case

Markdown in `displayName` would improve UX of dashboard, for example it would be possible to specify link to PR, commit, highlight some information, etc. 
Good example of it can be found in [Argo Workflows](https://argo-workflows.readthedocs.io/en/latest/title-and-description/#embedded-markdown).

**Comment by AlanGreene** [comment]
Thanks for the feature request @say5. That's something we may consider as part of https://github.com/tektoncd/dashboard/issues/2306

I'm not sure `displayName` is the correct place to include URLs or other potentially large values, that seems better suited to a `description` field or similar. However, this is something we can explore as part of the redesign of the PipelineRun and TaskRun details pages.

Apart from the examples in the Argo docs you provided above, do you have concrete examples / use cases for how you plan to use such a feature? Having specific use cases really helps to understand the problem and ensure we provide the correct solution.

**Comment by say5** [comment]
@AlanGreene in our use case we mostly trigger PipelineRuns from GitHub, in other CI/CD systems you have a page with all builds you have with basic primary information in one place - author on changes, link to PR, branch name, etc. Currently it is not possible with Tekton/Dashboard PipelineRuns page  (no `displayName` in PipelineRun). We have `displayName` in `pipelinerun.tasks` so we are using it to put all this info to the first task in PipelineRun, far from ideal but better than nothing, still there is no way to make links work, also in UI it shows rather limited string.

Here is an example of similar issue: https://github.com/tektoncd/dashboard/issues/3323

Regarding `task.description` - afaik there is no variable substitution for it, as it is just general description of a task and idea is to use substitution to display dynamic information about "build". In my view two changes can improve UX a lot:
1. [displayName for PipelineRun](https://github.com/tektoncd/community/pull/1174).
2. Markdown support to allow links, size, style, emojis, etc.

**Comment by AlanGreene** [comment]
> Regarding task.description - afaik there is no variable substitution for it, as it is just general description of a task

That is correct. I wasn't referring to that field specifically, rather pointing out that a field to provide a longer form description may be more suitable for some use cases than the display name, which ideally should be kept relatively short.

Thanks for the additional detail, it's very helpful. I think this would make sense to consider after PipelineRun displayName is supported in Tekton Pipelines.
